### PR TITLE
New 1762 map typcheck error

### PR DESCRIPTION
**Pull request by zainen**
have not recreated a failing test yet though 

Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: packages/taquito/test/contract/rpc-contract-provider.spec.ts [hunk 1]
```diff
@@ -1856,4 +1856,233 @@ describe('RpcContractProvider test', () => {
       done();
     });
   });
+
+  describe('getSaplingDiffByID', () => {
```
Comment (roxaneletourneau) [inline packages/taquito/test/contract/rpc-contract-provider.spec.ts:1860]: This test and the following seem to be duplicated.

File: packages/taquito/test/contract/rpc-contract-provider.spec.ts [hunk 2]
```diff
@@ -1856,4 +1856,233 @@ describe('RpcContractProvider test', () => {
...
+      done();
+    });
+  });
+
+  describe('Testing Bug #1762 mapTypecheckError', () => {
+    it('should have a defined response and storage of 3 should not throw an error', async (done) => {
+      mockRpcClient.getEntrypoints.mockResolvedValue({
+        entrypoints: {},
+      })
+      mockRpcClient.getContract.mockResolvedValue({
```
Comment (roxaneletourneau) [inline packages/taquito/test/contract/rpc-contract-provider.spec.ts:1907]: This mocked value takes considerable space in the test. Consider adding it to a separate file and importing it from there.

Comment (zainen) [reply packages/taquito/test/contract/rpc-contract-provider.spec.ts:1907]: 👍

File: packages/taquito/test/contract/rpc-contract-provider.spec.ts [hunk 3]
```diff
@@ -1856,4 +1857,20 @@ describe('RpcContractProvider test', () => {
       done();
     });
   });
+
+  describe('Testing Bug #1762 mapTypecheckError', () => {
```
Comment (dsawali) [inline packages/taquito/test/contract/rpc-contract-provider.spec.ts:1861]: Try and use the name of the member method/variable as the describe block name. In this case, it should be something like `describe('storage', ...`

describe blocks should be quite high level and general sounding

File: packages/taquito/test/contract/rpc-contract-provider.spec.ts [hunk 4]
```diff
@@ -1856,4 +1857,20 @@ describe('RpcContractProvider test', () => {
...
+    it('should have a defined response and storage of 3 should not throw an error', async (done) => {
```
Comment (dsawali) [inline packages/taquito/test/contract/rpc-contract-provider.spec.ts:1862]: you can put the issue # in the it block like so:
```
it('should return storage without errors (#1762)', ...
```

or even a comment above the `it` block linking to the issue itself

**Review by dsawali — CHANGES_REQUESTED** [review]
Could we also get a negative case unit test in for what would make a `.storage` call fail?

**Review by roxaneletourneau — APPROVED** [review]
Lgtm! :)

**Review by dsawali — APPROVED** [review]

**Pull request merged**